errors.go

```diff
@@ -0,0 +1,18 @@
+// Package gluon implements an IMAP4rev1 (+ extensions) mailserver.
+package gluon
+
+import (
+	"errors"
+
+	"github.com/ProtonMail/gluon/internal/state"
+)
+
+// IsNoSuchMessage returns true if the error is ErrNoSuchMessage.
+func IsNoSuchMessage(err error) bool {
```

Comment:
Can you verify this with at test. I have a suspicion that this will not work as we are reporting ent errors to the connector.

Reply:
done, it works as expected